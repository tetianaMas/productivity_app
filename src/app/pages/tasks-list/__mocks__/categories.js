export const categories = {
  work: { tasks: [], title: 'work' },
  education: { tasks: [], title: 'education' },
  hobby: {
    tasks: [
      {
        categoryId: 'hobby',
        completeDate: '03-17-21',
        createDate: { day: 17, fullDeadline: 1615978347385, month: 'March' },
        deadline: 'March 26, 2021',
        deadlineDate: {
          day: '',
          fullDeadline: 1616716800000,
          month: 'today',
        },
        description: 'Fgdgerf',
        estimation: '5',
        failedPomodoros: [(0, 1, 2, 3, 4)],
        id: '00807824925639886039',
        priority: 'high',
        status: 'COMPLETED',
        title: 'Ывпыпа',
      },
    ],
    title: 'hobby',
  },
  sport: { tasks: [], title: 'sport' },
  other: { tasks: [], title: 'other' },
};
