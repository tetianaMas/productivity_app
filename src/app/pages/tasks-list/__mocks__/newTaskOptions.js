export const newTaskOptions = {
  categoryId: 'other',
  completeDate: null,
  completedCount: [],
  createDate: { fullDeadline: 1617482733224, month: 'April', day: 3 },
  deadline: 'April 5, 2021',
  deadlineDate: { fullDeadline: 1617570000000, month: 'April', day: 5 },
  description: 'Hello! Let`s start cleaning!',
  estimation: '5',
  failedPomodoros: [],
  id: '61500269640882320691',
  priority: 'middle',
  status: 'GLOBAL_LIST',
  title: 'Test',
};
