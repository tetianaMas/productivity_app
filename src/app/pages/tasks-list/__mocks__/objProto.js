export const objProto = {
  work: {},
  education: {},
  hobby: {},
  sport: {},
  other: {},
};
