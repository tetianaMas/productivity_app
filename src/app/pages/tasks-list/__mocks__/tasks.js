export const tasks = [
  {
    categoryId: 'hobby',
    completeDate: '03-17-21',
    createDate: { day: 17, fullDeadline: 1615978347385, month: 'March' },
    deadline: 'March 26, 2021',
    deadlineDate: { day: '', fullDeadline: 1616716800000, month: 'today' },
    description: 'Fgdgerf',
    estimation: '5',
    failedPomodoros: [(0, 1, 2, 3, 4)],
    id: '00807824925639886039',
    priority: 'high',
    status: 'COMPLETED',
    title: 'Ывпыпа',
  },
  {
    categoryId: 'hobby',
    createDate: { day: 2, fullDeadline: 1617314051232, month: 'April' },
    deadline: 'April 15, 2021',
    deadlineDate: { day: 15, fullDeadline: 1618434000000, month: 'April' },
    description: 'Ergrtghrtg',
    estimation: '3',
    id: '08835933769418409683',
    priority: 'urgent',
    status: 'GLOBAL_LIST',
    title: 'Fgbrg',
  },
];
