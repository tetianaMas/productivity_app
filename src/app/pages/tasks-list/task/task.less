@import (reference) '~@less-helpers-module/_mixins.less';
@import (reference) '~@less-helpers-module/_variables.less';

.main {
  .task {
    display: flex;
    max-height: 6.25em;
    position: relative;
    margin-top: 12/@rem;
    background: @white-color;
    box-shadow: 0 5/@rem 8.3/@rem rgba(34, 56, 73, 0.302);

    &__category {
      width: 8/@rem;
      max-width: 100%;
      position: absolute;
      height: 100%;

      &--hobby {
        .category-color(@hobby-category-color, @hobby-category-active);
      }

      &--work {
        .category-color(@work-category-color, @work-category-active);
      }

      &--education {
        .category-color(@education-category-color, @education-category-active);
      }

      &--sport {
        .category-color(@sport-category-color, @sport-category-active);
      }

      &--other {
        .category-color(@other-category-color, @other-category-active);
      }

      &--delete {
        width: 40/@rem;
        max-width: 100%;
        padding: 0 24/@vw;
        .flex-center();
        transition: 0.2s;

        &.task__category--work {
          .evaluation(@work-category-color, @work-category-active);
        }

        &.task__category--education {
          .evaluation(@education-category-color, @education-category-active);
        }

        &.task__category--hobby {
          .evaluation(@hobby-category-color, @hobby-category-active);
        }

        &.task__category--sport {
          .evaluation(@sport-category-color, @sport-category-active);
        }

        &.task__category--other {
          .evaluation(@other-category-color, @other-category-active);
        }

        .evaluation();

        &~.task__estimation {
          pointer-events: none;
        }
      }

      &__icon {
        color: @white-color;
        font-size: 40/@rem;
        animation: delete-icon-appearing 1s;
      }

      @keyframes delete-icon-appearing {
        0% {
          opacity: 0.1;
          transform: rotateY(90deg);
        }
  
        100% {
          opacity: 1;
          transform: rotateY(0deg);
        }
      }  
    }

    &__deadline {
      max-width: 100%;
      padding: 0 31/@vw 0 39/@vw;
      .flex-center();
      color: @dark-gray-color;
      .cancel-user-selection();

      &__title {
        #fonts>.heading2(11/@rem, 16/@rem, 700, uppercase);
        color: inherit;
      }

      &--hidden {
        display: none;
      }
    }

    &__info {
      max-width: 100%;
      width: 396/@rem;
      padding: 16/@rem 0;
      flex-grow: 2;
    }

    &__title {
      #fonts>.heading3();
      .cancel-user-selection();

      &--urgent {
        color: @priority-urgent-title;
      }

      &--high {
        color: @priority-high-title;
      }

      &--middle {
        color: @priority-middle-title;
      }

      &--low {
        color: @priority-low-title;
      }

      &--done {
        text-decoration: line-through;
      }
    }

    &__text {
      margin-top: 4/@rem;
      max-height: 60%;
      #fonts>.paragraph();
      color: @grey-color;
      overflow: hidden;
      text-overflow: ellipsis;
      .cancel-user-selection();
    }

    &__controls {
      width: 92/@rem;
      padding: 20/@vw;
      max-width: 100%;
      .flex-center();
      justify-content: space-around;

      &__btn {
        position: relative;
        margin-right: 16/@rem;
        .btn-link();

        &:hover {
          &:before {
            color: @nepal-color;
          }
        }

        &:focus {
          &:before {
            color: @nepal-color;
          }
        }

        &--hidden {
          visibility: hidden;
        }

        &:before {
          font-size: 20/@rem;
          color: lighten(@silver-color, 0.9%);
        }
      }

      &--hidden {
        visibility: hidden;
      }

      &>button {
        display: flex;
      }
    }

    &__estimation {
      padding: 20/@vw;
      .flex-center();

      &__link {
        position: relative;

        &::before {
          font-size: 48/@rem;
          color: @white-color;
        }
      }

      &__number {
        position: absolute;
        transform: translate(-50%, -50%);
        left: 50%;
        bottom: 0;
        #fonts>.caption();
        color: @white-color;
      }

      &--urgent {
        .evaluation(@priority-urgent-color, @priority-urgent-active);
      }

      &--high {
        .evaluation(@priority-high-color, @priority-high-active);
      }

      &--middle {
        .evaluation(@priority-middle-color, @priority-middle-active);
      }

      &--low {
        .evaluation(@priority-low-color, @priority-low-active);
      }

      &--done {
        .opacity(0.8);
      }
    }

    &--done {
      background: rgba(255, 255, 255, 0.8);

      .task__estimation {
        pointer-events: none;
      }
    }

    &--hidden {
      display: none;
    }
  }

  .global-list {
    .task {
      &__category {
        &--work {
          .task-mark(@work-category-color);
        }

        &--education {
          .task-mark(@education-category-color);
        }

        &--sport {
          .task-mark(@sport-category-color);
        }

        &--hobby {
          .task-mark(@hobby-category-color);
        }

        &--other {
          .task-mark(@other-category-color);
        }
      }

      &__deadline {

        &--date {
          flex-direction: column;

          &--numeric {
            margin-bottom: 8/@rem;
            #fonts>.caption(28/@rem, 16/@rem);
            color: inherit;
          }
        }

        &--overdue {
          color: @priority-urgent-title;
        }
      }

      &__estimation {
        pointer-events: none;
      }
    }
  }
}

.task-list-mobile();
.task-list-tablet();
.task-list-desktop();