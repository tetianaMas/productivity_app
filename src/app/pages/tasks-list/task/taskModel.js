/**
 * @description model of task
 * @exports TaskModel
 * @class TaskModel
 */
export class TaskModel {
  /**
   * @description Creates an instance of TaskModel
   * @param {object} options task options
   * @memberof TaskModel
   */
  constructor(options) {
    this.id = options.id;
    this.title = options.title;
    this.description = options.description;
    this.createDate = options.createDate;
    this.deadlineDate = options.deadlineDate;
    this.failedPomodoros = options.failedPomodoros;
    this.completeDate = options.completeDate;
    this.completedCount = options.completedCount;
    this.priority = options.priority;
    this.categoryId = options.categoryId;
    this.status = options.status;
    this.estimation = options.estimation;
    this.deadline = options.deadline;
    this.options = options;
  }
}
