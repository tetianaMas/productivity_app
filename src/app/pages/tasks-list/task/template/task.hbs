  {{#if completeDate}}
  <div class="task js-task task--done" id="{{id}}">
    {{else}}
    <div class="task js-task" id="{{id}}">
  {{/if}}
    <div class="task__category task__category--{{categoryId}} js-task__category">
    </div>
    {{#if completeDate}}
    <div class="task__deadline task__deadline--date">
      {{else}}
    <div class="task__deadline task__deadline--date {{checkOverdue deadlineDate.fullDeadline}}">
    {{/if}}
      {{#if deadlineDate.day}}
      <span class="task__deadline--date--numeric">{{deadlineDate.day}}</span>
      {{/if}}
      {{#if completeDate}}
      <span class="task__deadline--date--numeric">{{getDate completeDate 'day'}}</span>
      <span class="task__deadline__title">{{getDate completeDate 'month'}}</span>
      {{else}}
      <span class="task__deadline__title">{{deadlineDate.month}}</span>
      {{/if}}
    </div>

    <div class="task__info">
      <div>
        {{#if completeDate}}
        <h2 class="task__title task__title--{{priority}} task__title--done">{{title}}</h2>
        {{else}}
        <h2 class="task__title task__title--{{priority}}">{{title}}</h2>
        {{/if}}
      </div>
      <div class="task__text">
        <p>{{description}}</p>
      </div>
    </div>

    <div class="task__controls js-btn-controls">
      {{#if completeDate}}
      
      {{else}}
      <button class="task__controls__btn {{checkButtonHidden status}} icon-arrows-up task__controls__btn--panel js-task-btn-up" title="Drag to Active">
      </button>

      <button class="task__controls__btn icon-edit task__controls__btn--panel js-task-btn-edit" title="Edit Task">
      </button>

      <button class="task__controls__btn icon-trash task__controls__btn--panel js-task-btn-delete" title="Delete Task">
      </button>
      {{/if}}
    </div>

    {{#if completeDate}}
    <div class="task__estimation task__estimation--{{priority}} js-link-start-task task__estimation--done">
      {{else}}
      <div class="task__estimation task__estimation--{{priority}} js-link-start-task">
    {{/if}}
      <button class="task__estimation__link icon-tomato icon-timer" title="Go to Timer">
        <span class="task__estimation__number">{{estimation}}</span>
      </button>
    </div>
  </div>
 