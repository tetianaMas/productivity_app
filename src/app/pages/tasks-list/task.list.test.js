/* eslint-disable no-undef */
import { TaskCollectionModel } from './MVC/taskCollectionModel';
import { dataService } from '../../services/dataService/dataService';
import allTasksDone from './template/all-tasks-done.hbs';
import emptyPage from './template/empty-page-message.hbs';
import firstPage from './template/firstPage.hbs';
import noDailyTasks from './template/no-daily-tasks-message.hbs';
import noTasksLeft from './template/no-tasks-left.hbs';
import taskListContainer from './template/task-list-container.hbs';
import { tasks } from './__mocks__/tasks';
import { tasksFiltered } from './__mocks__/tasksFiltered';
import { categories } from './__mocks__/categories';
import { objProto } from './__mocks__/objProto';
import { newTaskOptions } from './__mocks__/newTaskOptions';
import { TaskModel } from './task/taskModel';

const taskListModel = new TaskCollectionModel(dataService);
taskListModel.taskListCollection = tasks;

describe('rendering', () => {
  test('renders correctly allTasksDone', () => {
    const tree = JSON.stringify(allTasksDone());
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly emptyPage', () => {
    const tree = JSON.stringify(emptyPage());
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly firstPage', () => {
    const tree = JSON.stringify(firstPage(taskListModel.taskListTemplateData));
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly noDailyTasks', () => {
    const tree = JSON.stringify(noDailyTasks());
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly noTasksLeft', () => {
    const tree = JSON.stringify(noTasksLeft());
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly taskListContainer', () => {
    const tree = JSON.stringify(
      taskListContainer(taskListModel.taskListTemplateData)
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('createTaskListCollection', () => {
  test('should create task-list', () => {
    taskListModel.createTaskListCollection();
    expect(taskListModel.tasksCollection).toEqual(tasksFiltered);
  });
});

describe('createCategories', () => {
  test('should create categories', () => {
    const res = taskListModel.createCategories([tasks[0]], objProto);
    expect(res).toEqual(categories);
  });

  test('should return falsy result', () => {
    const res = taskListModel.createCategories([tasks[0]]);
    expect(res).toBeFalsy();
  });

  test('should return falsy result', () => {
    const res = taskListModel.createCategories();
    expect(res).toBeFalsy();
  });

  test('should return falsy result', () => {
    const res = taskListModel.createCategories('test', true);
    expect(res).toBeFalsy();
  });
});

describe('createTask', () => {
  test('should create new task', () => {
    spyOn(taskListModel, 'createTaskListCollection');
    spyOn(taskListModel, 'sendTaskData');

    const res = new TaskModel(newTaskOptions);
    taskListModel.createTask(newTaskOptions);
    expect(
      taskListModel.taskListCollection[
        taskListModel.taskListCollection.length - 1
      ]
    ).toEqual(res.options);
    expect(taskListModel.createTaskListCollection).toBeCalled();
    expect(taskListModel.sendTaskData).toBeCalled();
  });

  test('should return falsy result', () => {
    const res = taskListModel.createTask();
    expect(res).toBeFalsy();
  });

  test('should return falsy result', () => {
    const res = taskListModel.createTask(2343434);
    expect(res).toBeFalsy();
  });

  test('should return falsy result', () => {
    const res = taskListModel.createTask(true);
    expect(res).toBeFalsy();
  });
});

describe('getCurrentDate', () => {
  test('should return current day obj', () => {
    const res = taskListModel.getCurrentDate();
    expect(res).toBeTruthy();
  });

  test('should return day obj', () => {
    const today = new Date(2021, 2, 26);
    const options = {
      month: 'long',
    };
    const newDate = {
      fullDeadline: today.getTime(),
      month: Intl.DateTimeFormat('en-US', options).format(today),
      day: today.getDate(),
    };

    const res = taskListModel.getCurrentDate('March 26, 2021');
    expect(newDate).toEqual(res);
  });

  test('should return current day obj', () => {
    const res = taskListModel.getCurrentDate(true);
    expect(res).toBeTruthy();
  });

  test('should return current day obj', () => {
    const res = taskListModel.getCurrentDate('');
    expect(res).toBeTruthy();
  });
});

describe('updateTask', () => {
  test('should update current task', () => {
    spyOn(taskListModel, 'createTaskListCollection');
    spyOn(taskListModel, 'sendTaskData');

    const newTask = tasks[1];
    const newDescription = 'Wow! This is the greatest test ever!!!';
    newTask.description = newDescription;
    taskListModel.updateTask(newTask);
    const updatedTask = taskListModel.taskListCollection.find(
      item => item.id === newTask.id
    );
    expect(updatedTask.description).toBe(newDescription);
    expect(taskListModel.createTaskListCollection).toBeCalled();
    expect(taskListModel.sendTaskData).toHaveBeenCalledWith(
      'tasks',
      newTask,
      newTask.id
    );
  });

  test('should return falsy value', () => {
    const res = taskListModel.updateTask();
    expect(res).toBeFalsy();
  });
});

describe('changeTaskStatus', () => {
  test('should change task status', () => {
    spyOn(taskListModel.dataService, 'sendData');

    const id = tasks[0].id;
    const newStatus = 'TEST';
    taskListModel.changeTaskStatus(id, newStatus);
    const updatedTask = taskListModel.taskListCollection.find(
      item => item.id === id
    );
    expect(updatedTask.status).toBe(newStatus);
    expect(taskListModel.dataService.sendData).toHaveBeenCalledWith(
      'tasks',
      updatedTask,
      id
    );
  });

  test('should return falsy value', () => {
    const res = taskListModel.changeTaskStatus();
    expect(res).toBeFalsy();
  });
});

describe('changeTaskStatus', () => {
  test('should change task status', () => {
    spyOn(taskListModel.dataService, 'removeItem');
    spyOn(taskListModel.renderNoTasksLeftPageEvent, 'notify');

    const idsToRemove = [];
    taskListModel.taskListCollection.forEach(task => idsToRemove.push(task.id));
    taskListModel.removeTask(idsToRemove);
    expect(taskListModel.taskListCollection.length).toBeFalsy();
    expect(taskListModel.dataService.removeItem).toHaveBeenCalledWith(
      'tasks',
      idsToRemove
    );
    expect(taskListModel.renderNoTasksLeftPageEvent.notify).toBeCalled();
  });

  test('should return falsy value', () => {
    const res = taskListModel.removeTask();
    expect(res).toBeFalsy();
  });
});

describe('checkLeftTasks', () => {
  test('should call appropriate func', () => {
    spyOn(taskListModel.renderNoTasksLeftPageEvent, 'notify');

    taskListModel.checkLeftTasks();
    expect(taskListModel.renderNoTasksLeftPageEvent.notify).toBeCalled();
  });

  test('should call appropriate func', () => {
    spyOn(taskListModel.renderEmptyDailyTasks, 'notify');

    taskListModel.taskListCollection.length = 1;
    taskListModel.globalTasksActive.length = 1;
    taskListModel.checkLeftTasks();
    expect(taskListModel.renderEmptyDailyTasks.notify).toBeCalled();
  });

  test('should call appropriate func', () => {
    spyOn(taskListModel.renderAddFirstTaskPageEvent, 'notify');

    taskListModel.taskListCollection.length = 1;
    taskListModel.globalTasksActive.length = 0;
    taskListModel.dailyTasksActive.length = 0;
    taskListModel.dailyTasksCompleted.length = 0;
    taskListModel.globalTasksCompleted.length = 0;

    taskListModel.checkLeftTasks();
    expect(taskListModel.renderAddFirstTaskPageEvent.notify).toBeCalled();
  });

  test('should call appropriate func', () => {
    spyOn(taskListModel.renderNoTasksLeftPageEvent, 'notify');

    taskListModel.taskListCollection.length = 1;
    taskListModel.globalTasksActive.length = 0;
    taskListModel.dailyTasksActive.length = 0;
    taskListModel.dailyTasksCompleted.length = 1;
    taskListModel.globalTasksCompleted.length = 1;

    taskListModel.checkLeftTasks();
    expect(taskListModel.renderNoTasksLeftPageEvent.notify).toBeCalled();
  });
});

describe('getTaskById', () => {
  beforeAll(() => {
    taskListModel.taskListCollection = tasks;
  });
  test('should return task by id', () => {
    const res = taskListModel.getTaskById(tasks[0].id);
    expect(res).toEqual(tasks[0]);
  });

  test('should return falsy value', () => {
    const res = taskListModel.getTaskById();
    expect(res).toBeFalsy();
  });
});

describe('saveTaskToStorage', () => {
  beforeAll(() => {
    taskListModel.taskListCollection = tasks;
  });
  test('should call func to set value to storage', () => {
    spyOn(taskListModel.dataService, 'setDataToStorage');
    taskListModel.saveTaskToStorage(tasks[0].id);
    expect(taskListModel.dataService.setDataToStorage).toHaveBeenCalledWith(
      'active-task',
      tasks[0]
    );
  });

  test('should return falsy value', () => {
    const res = taskListModel.saveTaskToStorage();
    expect(res).toBeFalsy();
  });
});
