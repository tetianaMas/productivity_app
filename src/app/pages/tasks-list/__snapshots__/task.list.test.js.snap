// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly allTasksDone 1`] = `"\\"<div class=\\\\\\"main__content--message\\\\\\">\\\\r\\\\n    <div>\\\\r\\\\n      <p>Excellent,</p>\\\\r\\\\n      <p>all daily tasks done :)</p>\\\\r\\\\n    </div>\\\\r\\\\n</div>\\""`;

exports[`rendering renders correctly emptyPage 1`] = `"\\"<div class=\\\\\\"first-screen\\\\\\">\\\\r\\\\n    <img src=\\\\\\"../images/tomato-add.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" class=\\\\\\"first-screen__logo\\\\\\">\\\\r\\\\n    <div class=\\\\\\"first-screen__message\\\\\\">\\\\r\\\\n      <p>Add your first task</p>\\\\r\\\\n    </div>\\\\r\\\\n</div>\\""`;

exports[`rendering renders correctly firstPage 1`] = `"\\" <header class=\\\\\\"header js-header-main\\\\\\">\\\\r\\\\n\\\\r\\\\n <h1 class=\\\\\\"header__title\\\\\\">\\\\r\\\\n    daily task list\\\\r\\\\n  </h1>\\\\r\\\\n  <div class=\\\\\\"header__logo d-none\\\\\\">\\\\r\\\\n    <a href=\\\\\\"#\\\\\\">\\\\r\\\\n      <img src=\\\\\\"../images/Logo.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" />\\\\r\\\\n    </a>\\\\r\\\\n  </div>\\\\r\\\\n  <nav class=\\\\\\"navigation\\\\\\" aria-label=\\\\\\"Main navigation\\\\\\">\\\\r\\\\n    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-list \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-statistics \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-settings \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n    </ul>\\\\r\\\\n  </nav>\\\\r\\\\n</header><main class=\\\\\\"main\\\\\\">\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right \\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab \\\\\\" ></button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" ></button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n<div class=\\\\\\"first-screen\\\\\\">\\\\r\\\\n  <img\\\\r\\\\n    src=\\\\\\"../images/tomato_settings.svg\\\\\\"\\\\r\\\\n    alt=\\\\\\"Logo\\\\\\"\\\\r\\\\n    class=\\\\\\"first-screen__logo\\\\\\"\\\\r\\\\n  >\\\\r\\\\n  <div class=\\\\\\"first-screen__message\\\\\\">\\\\r\\\\n    <p>\\\\r\\\\n      As you visited site for a first time you can\\\\r\\\\n          check and customize your default\\\\r\\\\n          application settings\\\\r\\\\n    </p>\\\\r\\\\n  </div>\\\\r\\\\n  <div class=\\\\\\"first-screen__buttons\\\\\\">\\\\r\\\\n    <button class=\\\\\\"first-screen__button first-screen__button--blue js-btn-skip\\\\\\">\\\\r\\\\n      Skip\\\\r\\\\n    </button>\\\\r\\\\n    <button class=\\\\\\"first-screen__button first-screen__button--mint js-btn-to-settings\\\\\\">\\\\r\\\\n      Go to settings\\\\r\\\\n    </button>\\\\r\\\\n  </div>\\\\r\\\\n</div>\\\\r\\\\n</main>\\""`;

exports[`rendering renders correctly noDailyTasks 1`] = `"\\"<div class=\\\\\\"main__content--message\\\\\\">\\\\r\\\\n    <div>\\\\r\\\\n      <p>Task added,</p>\\\\r\\\\n      <p>Go it to the top 5 in daily task list</p>\\\\r\\\\n      <span class=\\\\\\"main__content__icon icon-arrow_circle\\\\\\"></span>\\\\r\\\\n    </div>\\\\r\\\\n</div> \\""`;

exports[`rendering renders correctly noTasksLeft 1`] = `"\\"<div class=\\\\\\"first-screen\\\\\\">\\\\r\\\\n    <img src=\\\\\\"../images/tomato-addv02.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" class=\\\\\\"first-screen__logo first-screen__logo--width--200\\\\\\">\\\\r\\\\n      <div class=\\\\\\"first-screen__message\\\\\\">\\\\r\\\\n        <p>You donâ€™t have any tasks left.</p>\\\\r\\\\n        <p>Add new tasks to stay productive.</p>\\\\r\\\\n      </div>\\\\r\\\\n</div>\\""`;

exports[`rendering renders correctly taskListContainer 1`] = `"\\" <header class=\\\\\\"header js-header-main\\\\\\">\\\\r\\\\n\\\\r\\\\n <h1 class=\\\\\\"header__title\\\\\\">\\\\r\\\\n    daily task list\\\\r\\\\n  </h1>\\\\r\\\\n  <div class=\\\\\\"header__logo d-none\\\\\\">\\\\r\\\\n    <a href=\\\\\\"#\\\\\\">\\\\r\\\\n      <img src=\\\\\\"../images/Logo.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" />\\\\r\\\\n    </a>\\\\r\\\\n  </div>\\\\r\\\\n  <nav class=\\\\\\"navigation\\\\\\" aria-label=\\\\\\"Main navigation\\\\\\">\\\\r\\\\n    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-list \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-statistics \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-settings \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n    </ul>\\\\r\\\\n  </nav>\\\\r\\\\n</header>\\\\r\\\\n<main class=\\\\\\"main\\\\\\">\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right \\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab \\\\\\" ></button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" ></button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n    <div class=\\\\\\"main__content--daily main__content--active js-main__content--daily js-container-daily js-tabcontent\\\\\\" aria-label=\\\\\\"Daily tasks List\\\\\\"></div>\\\\r\\\\n    <div class=\\\\\\"main__content--done js-main__content--done d-none js-tabcontent\\\\\\" aria-label=\\\\\\"Done tasks List\\\\\\"></div>\\\\r\\\\n\\\\r\\\\n    <ol class=\\\\\\"global-list d-none js-global-list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"global-list__nav js-nav\\\\\\">\\\\r\\\\n        <div>\\\\r\\\\n          <button class=\\\\\\"global-list__title js-global-list-btn-toggler\\\\\\" title=\\\\\\"Go to Global List\\\\\\">\\\\r\\\\n            Global list\\\\r\\\\n            <span class=\\\\\\"global-list__icon icon-global-list-arrow-right js-global-list-open-close-icon\\\\\\"></span>\\\\r\\\\n          </button>\\\\r\\\\n\\\\r\\\\n          <div class=\\\\\\"tabs js-global-tabs\\\\\\" aria-label=\\\\\\"Tasks select navigation\\\\\\">\\\\r\\\\n            <div class=\\\\\\"tab tab--left tab--hidden js-tab-select\\\\\\">\\\\r\\\\n              <button class=\\\\\\"tab__btn js-select-global-list\\\\\\">Select all</button>\\\\r\\\\n              <button class=\\\\\\"tab__btn js-deselect-global-list\\\\\\">Deselect all</button>\\\\r\\\\n            </div>\\\\r\\\\n          </div>\\\\r\\\\n        </div>\\\\r\\\\n\\\\r\\\\n        <div class=\\\\\\"tabs tabs--filter js-global-tabs js-global-tabs-categories\\\\\\" aria-label=\\\\\\"Tasks priority navigation\\\\\\">\\\\r\\\\n          <div class=\\\\\\"tab\\\\\\">\\\\r\\\\n            <button class=\\\\\\"tab__btn js-tab-btn tab__btn--active\\\\\\">all</button>\\\\r\\\\n            <button class=\\\\\\"tab__btn js-tab-btn\\\\\\">urgent</button>\\\\r\\\\n            <button class=\\\\\\"tab__btn js-tab-btn\\\\\\">high</button>\\\\r\\\\n            <button class=\\\\\\"tab__btn js-tab-btn\\\\\\">middle</button>\\\\r\\\\n            <button class=\\\\\\"tab__btn js-tab-btn\\\\\\">low</button>\\\\r\\\\n          </div>\\\\r\\\\n        </div>\\\\r\\\\n      </li>\\\\r\\\\n\\\\r\\\\n      <li class=\\\\\\"js-global-list-container\\\\\\">\\\\r\\\\n        <div>\\\\r\\\\n          <div class=\\\\\\"global-list__content js-global-list-content js-container js-container-global js-tabcontent\\\\\\">\\\\r\\\\n          </div>\\\\r\\\\n\\\\r\\\\n          <div class=\\\\\\"global-list__content js-global-list-content--done js-container js-container-global d-none js-tabcontent\\\\\\">\\\\r\\\\n          </div>\\\\r\\\\n        </div>\\\\r\\\\n      </li>\\\\r\\\\n      \\\\r\\\\n    </ol>\\\\r\\\\n  </main>\\""`;
